modules/lang-painless/src/main/java/org/elasticsearch/painless/Definition.java

```diff
@@ -1011,7 +1011,8 @@ private void addMethod(ClassLoader whitelistClassLoader, String ownerStructName,
                     "and parameters " + whitelistMethod.painlessParameterTypeNames, iae);
         }
 
-        if (javaMethod.getReturnType() != defClassToObjectClass(painlessReturnClass)) {
+        // allow convariant return types in implementing classes.
+        if (!defClassToObjectClass(painlessReturnClass).isAssignableFrom(javaMethod.getReturnType())) {
```

Comment:
Had to fix this because i had to introduce a new sub class of ReadableDateTime and i wanted to change the return types from RDT to my new class.